micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/correlationid/CorrelationIdAspect.java

```diff
@@ -80,6 +90,17 @@ public Object call() throws Exception {
         });
     }
 
+    @Around("anyControllerOrRestControllerWithPublicWebAsyncTaskMethod()")
+    public Object wrapWebAsyncTaskWithCorrelationId(ProceedingJoinPoint pjp) throws Throwable {
+        final WebAsyncTask webAsyncTask = (WebAsyncTask) pjp.proceed();
+        log.debug("Wrapping webAsyncTask with correlation id [" + CorrelationIdHolder.get() + "]");
+
+        Field callableField = WebAsyncTask.class.getDeclaredField("callable");
```

Comment:
@Climbee I see why it can't implemented properly, too bad. Therefor I believe your implementation is the only possible right now. However can you think about wrapping your reflection code with `try`-`catch` and swallowing/logging exceptions? Then changes in underlying Spring implementation (e.g. field name changes) won't break micro-infra-spring, but just gracefully disable one feature. WDYT?